gnuradio-runtime/include/gnuradio/block.h

```diff
@@ -928,6 +930,59 @@ class GR_RUNTIME_API block : public basic_block
                             uint64_t rel_end,
                             const pmt::pmt_t& key);
 
+    /*!
+     * \brief Get the first tag in specified range (if any), fulfilling criterion
+     *
+     * \details
+     * This function returns the lowest-offset tag in the range for whom the predicate
+     * function returns true.
+     *
+     * The predicate function hence needs to map tags to booleans; its signature is
+     * bool function(const tag_t& tag_to check);
+     *
+     * A sensible choice is a side-effect-free lambda, e.g., you'd use this as:
+     *
+     * auto timestamp = get_first_tag_in_range(
+     *     0,                          // which input
+     *     nitems_read(0),             // start index
+     *     nitems_read(0) + something, // end
+     *     [this](const gr::tag_t& tag) {
+     *         return pmt::eqv(tag.key, d_time_tag) && !pmt::is_null(tag.value)
+     *     });
+     * if (timestamp) {
+     *     d_logger->info("got time tag {} at offset {}",
+     *                    timestamp.value.value,
```

Comment:
Is this just `timestamp.value` and `timestamp.offset`? Based on the QA code, I think so.

Reply:
no, it's `timestamp.value.value`, because `timestamp` is a `std::optional`, and you need to access its value (if any, hence the `if` conditional) via its `value` member, which then is a tag, which has a `value` member.

So sadly, no way I can see around this

Reply:
In Python, a function doesn't have a fixed return type, so the wrapped function either returns a `NoneType` or `gr.tag_t`; hence, no need for the `.value` from the optional. But that's pretty pythonic, so it feels natural, I'd say.

Reply:
That was my point ... the existing functions use `.value` and the new function uses `.value.field`, so it requires the caller to know about `std::optional` vs the old way, making the API a little asymmetric.

Reply:
FWIW, I don't have a great solution either, since `std::optional` is easier to use but requires a different coding style. I can just picture people moving code to the new function, and then having to add either the extra `.value` or creating a new variable so the existing code still works. Not a huge deal.